engine/unit.go

```diff
@@ -74,14 +74,15 @@ func (u *Unit) LaunchAfter(delay time.Duration, f func()) {
 // If f is executed, the unit will not shut down until after f returns.
 func (u *Unit) LaunchPeriodically(f func(), interval time.Duration, delay time.Duration) {
 	u.Launch(func() {
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
+
 		select {
 		case <-u.ctx.Done():
 			return
 		case <-time.After(delay):
 		}
 
-		ticker := time.NewTicker(interval)
-		defer ticker.Stop()
 		for {
 			select {
 			case <-u.ctx.Done():
```

Comment:
I would consider this arguably a bug, which is why I've made the fix directly to `engine.Unit`.

The problem here is that the first tick from a `time.Ticker` will not be sent until `interval` has passed. Thus, the first invocation of the given function `f` is not made until `delay + interval` has passed.

Therefore, if the random delay chosen in `PeerManager` is almost as long as `interval`, we may wait almost 2*interval before getting our second invocation of `f`, hence timing out in the unit test.